components/brave_wallet/browser/eth_json_rpc_controller.cc

```diff
@@ -244,6 +258,7 @@ void EthJsonRpcController::OnGetTransactionReceipt(
 // static
 void EthJsonRpcController::RegisterProfilePrefs(PrefRegistrySimple* registry) {
   registry->RegisterListPref(kBraveWalletCustomNetworks);
+  registry->RegisterStringPref(kBraveWalletCurrentChainId, std::string());
```

Comment:
just register the default value here instead of using fallback function

Reply:
the fallback is used not only to initialize pref, but also to provide fallback if user adds some broken network

Reply:
so why didn't we prevents users from adding broken networks?

Reply:
we have issue about it https://github.com/brave/brave-browser/issues/17640, hope we can add this soon

Reply:
what I'm trying to say is, you didn't check broken networks in this PR so basically this fallback only serves as getting default value